Client/Frontend/Browser/URLBarView.swift

```diff
@@ -297,20 +376,41 @@ class URLBarView: UIView, BrowserLocationViewDelegate, UITextFieldDelegate {
                     self.tabsButton.transform = CGAffineTransformMakeTranslation(self.tabsButton.frame.width + URLBarViewUX.URLBarCurveOffset, 0)
                     self.curveShape.transform = CGAffineTransformMakeTranslation(self.tabsButton.frame.width + URLBarViewUX.URLBarCurveOffset + URLBarViewUX.URLBarCurveBounceBuffer, 0)
 
+                    if self.showToolbar {
+                        self.forwardButton.transform = CGAffineTransformMakeTranslation(-2 * AppConstants.ToolbarHeight, 0)
+                        self.backButton.transform = CGAffineTransformMakeTranslation(-2 * AppConstants.ToolbarHeight, 0)
+                        self.shareButton.alpha = 0
+                        self.bookmarkButton.alpha = 0
+                    }
                 } else {
                     self.cancelButton.alpha = 0
                     self.tabsButton.transform = CGAffineTransformIdentity
                     self.cancelButton.transform = CGAffineTransformMakeTranslation(self.cancelButton.frame.width, 0)
                     self.curveShape.transform = CGAffineTransformIdentity
+
+                    if self.showToolbar {
+                        self.forwardButton.transform = CGAffineTransformIdentity
+                        self.backButton.transform = CGAffineTransformIdentity
+                        self.shareButton.alpha = 1
+                        self.bookmarkButton.alpha = 1
+                    }
                 }
                 self.layoutIfNeeded()
-                }, completion: { _ in
-                    self.tabsButton.hidden = editing
-                    self.cancelButton.hidden = !editing
+            }, completion: { _ in
+                self.tabsButton.hidden = editing
+                self.cancelButton.hidden = !editing
+                self.forwardButton.hidden = !self.showToolbar || editing
+                self.backButton.hidden = !self.showToolbar || editing
+                self.shareButton.hidden = !self.showToolbar || editing
+                self.bookmarkButton.hidden = !self.showToolbar || editing
             })
         } else {
             self.tabsButton.hidden = editing
             self.cancelButton.hidden = !editing
+            self.forwardButton.hidden = !showToolbar || editing
+            self.backButton.hidden = !showToolbar || editing
+            self.shareButton.hidden = !showToolbar || editing
+            self.bookmarkButton.hidden = !showToolbar || editing
```

Comment:
Ugh, this `if animated` is becoming really hard to read with these changes. One thing I noticed is that this block is identical to the `completion` block directly above it. It's also very similar to lines 357-362. Instead of duplicating all of this in 3 places, can you extract this to a helper method that takes a boolean and sets these accordingly?